source/DoorNode.cpp

```diff
@@ -0,0 +1,42 @@
+﻿#include "DoorNode.h"
+
+#include <cugl/2d/CUAnimationNode.h>
+
+using namespace cugl;
+
+/** The diameter of the ship. Also the x coordinate of the center of the ship */
+constexpr unsigned int DIAMETER = 1280;
+
+/** The radius used for placement of the doors. */
+constexpr unsigned int DOOR_RADIUS = 650;
+
+/** The height of the door. */
+int height = 0;
+
+void DoorNode::draw(const std::shared_ptr<cugl::SpriteBatch>& batch, const Mat4& transform,
+					Color4 tint) {
+	Vec2 doorPos = Vec2(DIAMETER + DOOR_RADIUS * sin(doorModel->getAngle()),
+						DIAMETER / 2.0f - (DOOR_RADIUS)*cos(doorModel->getAngle()));
+	if (doorModel->getAngle() < 0) {
+		doorPos = Vec2(0, 0);
+	}
+	setPosition(doorPos);
+	setAngle(doorModel->getAngle());
+	switch (doorModel->getPlayersOn()) {
+		case 0:
+			setFrame(0);
+			break;
+		case 1:
+			setFrame(1);
+			break;
+		default:
+			setFrame(2);
+			break;
+	}
```

Comment:
Is there a reason we can't just pass `getPlayersOn()` directly as an argument to `setFrame()`, maybe after clamping it to `2` max? This seems like a lot of code for a very simple thing.

Reply:
good catch, I left that in when I was considering opening the doors differently.